metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/util/CondUpdateUtil.java

```diff
@@ -0,0 +1,33 @@
+package io.datahubproject.openapi.util;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+@Slf4j
+public class CondUpdateUtil {
+
+    // condUpdate String format: "urn1=createdOn1;urn2=createdOn2;...;urnN=createdOnN"
+
+    public static Map<String, Long> extractCondUpdate(String condUpdate) {
+        Map<String, Long> createdOnMap = new HashMap<>();
+        if (condUpdate != null) {
+            Arrays.stream(condUpdate.split(";"))
+                    .forEach(item -> {
+                        String[] values = item.split("=");
+                        if (values.length == 2) {
+                            try {
+                                createdOnMap.put(values[0], Long.parseLong(values[1]));
```

Comment:
The official format of [If-Unmodified-Since](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Unmodified-Since) is really awful, but we should either support it or specifically call out the fact that we are deviating from the standard.

```
If-Unmodified-Since: <day-name>, <day> <month> <year> <hour>:<minute>:<second> GMT
```

We should summarise the change in this PR to include the scenarios we are catering for, and also link to the [Conditional Requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Conditional_requests#avoiding_the_lost_update_problem_with_optimistic_locking) document that is the inspiration for our design.